### ISSUE TITLE
[graphiql] `GraphiQL.visiblePlugin` breaks the Documentation Explorer

### ISSUE THREAD

**Issue by dylanowen** [graphql/graphiql#2790]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When setting the `GraphiQL.visiblePlugin` property, any navigation in the Documentation Explorer opens the "visible" plugin. Not the correct Documentation Explorer page.

### Expected Behavior

`visiblePlugin` shouldn't break the Documentation Explorer

### Steps To Reproduce

```Markdown
BUGIN = {
    title: 'Bug Example',
    icon: () => {
        return <div>Bug</div>;
    },
    content: () => <div>Bug Content</div>
}

const App = () => (
    <GraphiQL
        plugins={[BUGIN]}
        visiblePlugin={BUGIN}
    />
);
```

Then try navigating inside the Documentation Explorer

### Module pattern

- [ ] graphiql-umd
- [ ] graphiql-esm
- [ ] graphiql-commonjs

### Environment

```Markdown
- GraphiQL Version: 2.0.8
- OS: OSX 12.6
- Browser: Firefox 106 and Chrome 105
- Bundler: Webpack 5
- `react` Version: 17
- `graphql` Version: 16
```

**Comment by crjm** [comment]
Same issue here.

**Comment by thomasheyenbrock** [comment]
Hey @dylanowen 👋  this is the expected behavior. When using the `visiblePlugin` prop you're taking over the responsibility to control the visibility of the plugins. (There's currently no prop to define the initial value for the plugin visibility state if that's the behavior you're expecting.)

In order to get your example to work you would need to do something like this:
```js
import { useState } from 'react';

const BUGIN = {
    title: 'Bug Example',
    icon: () => <div>Bug</div>,
    content: () => <div>Bug Content</div>
};

const App = () => {
  const [visiblePlugin, setVisiblePlugin] = useState(BUGIN)
  return (
    <GraphiQL
      plugins={[BUGIN]}
      visiblePlugin={visiblePlugin}
      onTogglePluginVisibility={newVisiblePlugin => setVisiblePlugin(newVisiblePlugin)}
    />
  );
}
```

**Comment by thomasheyenbrock** [comment]
Hey @dylanowen 👋 I tried to reproduce this in a CodeSandbox but your example works just fine for me: https://codesandbox.io/s/eloquent-framework-uszlge?file=/src/App.tsx

Do you have a reproduction that you can share?

**Comment by thomasheyenbrock** [comment]
You're right! Sorry for that, I did not read the description thoroughly enough 🙈

**Comment by thomasheyenbrock** [comment]
Just released the fix for this with `@graphiql/react@0.13.3` and `graphiql@2.0.9`
